drivers/hmis/app/models/hmis/ce/opportunity.rb

```diff
@@ -67,8 +68,31 @@ class Opportunity < GrdaWarehouseBase
       scope
     }
 
+    scope :actives_first, -> {
+      o_t = Hmis::Ce::Opportunity.arel_table
+      order(
+        o_t[:status].when('closed').then(1).else(0).asc,
+        o_t[:created_at].desc,
+        o_t[:id].desc,
+      )
+    }
+
     def active?
       !closed?
     end
+
+    private
+
+    def unique_opportunity_per_unit
+      return if status.to_sym == :closed || owner.nil?
+
+      # This validator expects that owner's opportunities are preloaded, to avoid n+1 on save
+      conflicting_opportunity_exists = owner.opportunities.to_a.select do |existing_opp|
+        existing_opp.status != 'closed' && existing_opp.id != id
```

Comment:
do we want a to_sym or to_s here on `opp.status`, or are we sure its always a string?